### PR TITLE
Fix account recover-disable edge case

### LINKED ISSUE #3795: Fails to recover admin password on 1.7.2
### I did this

```
kubectl exec -it test-recreate-admin-passwords-default-0 -- kanidmd recover-account admin --output json
WARNING: No configuration path was provided, relying on environment variables.
00000000-0000-0000-0000-000000000000 INFO     ｉ [info]: Starting Kanidmd | version: 1.7.2
00000000-0000-0000-0000-000000000000 WARN     🚧 [warn]: This is running as uid == 0 (root) which may be a security risk.
00000000-0000-0000-0000-000000000000 WARN     🚧 [warn]: WARNING: DB folder /data has 'everyone' permission bits in the mode. This could be a security risk ...
00000000-0000-0000-0000-000000000000 INFO     ｉ [info]: Running account recovery ...
"error"
Logging pipeline completed shutdown
```

### I expected the following

Admin password recreated

### This happened instead

```
3e72bc67-6b0e-44bd-a235-fe009865506c INFO     handle_admin_client_request [ 12.0ms | 0.10% / 100.00% ]
3e72bc67-6b0e-44bd-a235-fe009865506c INFO     ┝━ handle_admin_recover_account [ 12.0ms | 99.90% ] name: "admin"
3e72bc67-6b0e-44bd-a235-fe009865506c ERROR    │  ┝━ 🚨 [error]: Ava len > 1 on single value attribute! | event_tag_id: 1
3e72bc67-6b0e-44bd-a235-fe009865506c ERROR    │  ┝━ 🚨 [error]: Schema Violation in validation of modify_pre_apply InvalidAttributeSyntax("account_valid_from") | event_tag_id: 1
3e72bc67-6b0e-44bd-a235-fe009865506c ERROR    │  ┕━ 🚨 [error]:  | err: SchemaViolation(InvalidAttributeSyntax("account_valid_from"))
3e72bc67-6b0e-44bd-a235-fe009865506c ERROR    ┕━ 🚨 [error]: error during recover-account | err: SchemaViolation(InvalidAttributeSyntax("account_valid_from"))
```

### Kanidm version details

- Output of `kanidm(d) version`: kanidmd 1.7.2
- Are you running it in a container? If so, which image/tag?: kanidm/server:1.7.2

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #3795 

Checklist

- [x] This PR contains no AI generated code
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/lib/src/idm/server.rs [hunk 1]
```diff
@@ -1934,6 +1935,7 @@ impl IdmServerProxyWriteTransaction<'_> {
         let modlist = ModifyList::new_list(vec![
             // Ensure that the account has no validity, and the expiry is now.
             m_purge(Attribute::AccountValidFrom),
+            m_purge(Attribute::AccountExpire),
             Modify::Present(Attribute::AccountExpire, v_expire),
```
Comment (yaleman) [inline server/lib/src/idm/server.rs:1939]: It seems silly to be removing then setting expiry here?

Comment (Firstyear) [reply server/lib/src/idm/server.rs:1939]: It's not silly - the modifies are stateful. If you set modify::present, it means "make sure that value exists". It DOESN'T clear existing values. That's why you need to remove then set.

Comment (yaleman) [reply server/lib/src/idm/server.rs:1939]: Ah, so present is an insert rather than an upsert?

Comment (Firstyear) [reply server/lib/src/idm/server.rs:1939]: Yep, it's just saying "assert that a value exists in the array" basicly. It doesn't remove anything already there.

Comment (Firstyear) [reply server/lib/src/idm/server.rs:1939]: We do have an upsert option in Modify but I didn't use it here.

**Review by yaleman — APPROVED** [review]

File: server/lib/src/idm/server.rs (hunk 1 above)
Comment (yaleman) [reply server/lib/src/idm/server.rs:1939]: Thanks for explaining it
